Add cookie login flow

yoshiemuranaka opened the pull request:
Add cookie based authentication

Closes #6

yoshiemuranaka committed:
Remove auth object from state

No need to store credentials.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Add cookie based login authentication

- Changed POST request data structure to match
  backend requirements for set-cookie in response
  header
- Added withCredentials property to default
  axios config
- Modifying proxied response to remove 'Secure' flag
  so browser can create Cookie while running locally

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka commented:
Not sure if this works for production/overlay. Can test with overlay when #38 is merged.

gtmills commented on src/store/modules/Authentication/AuthenticanStore.js:
```diff
@@ -3,52 +3,39 @@ import api from "../../api";
   getters: {
     authStatus: state => state.status,
-    isLoggedIn: state => !!state.token
+    isLoggedIn: (_, getters) => getters.authStatus === "authenicated"
```
spelling ? authenticated

yoshiemuranaka replied:
Thanks

yoshiemuranaka committed:
Fix spelling error in AuthenticationStore

Fix authStatus 'authenticated' spelling error.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Check for document cookie in authstore

Instead of checking for authorized state in store,
check for the document cookie to be set so user isn't
kicked off when doing a browser refresh.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Update method to remove 'Secure' attribute in config

Changed to use String replace method instead of split,
filter, join.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague reviewed (changes requested):
We found some issues on production that we are trying to resolve.

derick-montague commented:
I noticed that refreshing the page will redirect browser to login. Refreshing the page on login throws an error.

yoshiemuranaka committed:
Merge branch 'master' of github.com:gtmills/phosphor-webui-vue into i-6-auth

yoshiemuranaka committed:
Add user logout call

Fixes 405 issue when logging out and immediately refreshing
the login page.

- Add logout api request
- Add js-cookie package to manage browser cookies

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Update babel preset

Updated the babel preset config to include useBuiltIns,
which resolves MIME type errors when overlaying

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented on babel.config.js:
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-  presets: ["@vue/cli-plugin-babel/preset"]
+  presets: [["@babel/preset-env", { useBuiltIns: "entry", corejs: 3 }]]
```
We only need to change the useBuiltIns value. By default, vue cli-plugin-babel/preset sets the useBuiltIns to  `usage`. If we just update it to use `entry`, we are good. 

`  presets: [["@vue/cli-plugin-babel/preset", { useBuiltIns: "entry" }]]`

yoshiemuranaka replied:
Cool, amended my last commit

yoshiemuranaka committed:
Disable vue-router history mode

Getting 404 errors when refreshing certain pages. This
is expected behavior with history mode enabled unless
server updates are made. In the meantime, will use default
vue router hash mode.

https://router.vuejs.org/guide/essentials/history-mode.html

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague reviewed (approved):
Works great! Thank you, @yoshiemuranaka

derick-montague commented:
Integrated into webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28638
